### DISCUSSION TITLE
COMMENTS HERE

### DISCUSSION THREAD

**Discussion by tamlok** [tamlok/vnote#170] [Ideas]
Hi,

Having been developing VNote for one and a half years, I want to hear more comments about VNote. I am too lazy to set up a forum, so I post this issue to ask for your voice about VNote. Thanks!

P.S. Please do not put issues here. :)

**Comment by eda53** [comment]
Though I just tried out VNote last night, I feel so regret that I know it too late. It is my favourite software, and will be the one in years -- pretty sure!

Not sure whether the ordinary user would like it or not, but I suppose, as a vim user and a developer, the features of VNote are what I have being seeking for a long time: such as the markdown for document, syntax highlight for code as the note, the attachments as the collection of the various materials, magic word, etc.

By the way, I'd like to recommend a combination of VNote and Psimo Mount's personal folder which can store/compress/encrypt the folders used by VNote into a single file. So far for me, no one can beat this combination.

**Comment by vmexia** [comment]
Well I said it before and I will say it again, this piece of software is one my favorites since I read about it on Hacker News. For me, VNote + PowerGREP/GREP + GDrive is THE KILLER COMBO (Note that VNote already has the search build in, but I love the versatility of being able to just grep a bunch of files and be able to get results, unlike Microsoft's OneNote and other similar note taking tools).  

Keep up the awesome work :)

**Comment by tamlok** [comment]
@vmexia Thanks! VNote's built-in search also lists the matching lines. You could try Universal Entry for a quick search. It is crazy convenient!

**Comment by vmexia** [comment]
What I meant was that even without VNote installed your notes are still useful due to its open markdown format, unlike similar tools. Example: I have my notebooks synchronized with GDrive so I can easily read them anywhere I want, even without having VNotes installed, like on my phone/tablet. If I want to find a particular note and for some reason I don't happen to have VNote there I can just use a simple grep and VOILÀ. Obviously you will lose some of the built in functionality like the image preview etc, but even that is easily bypassed since you have the image path written on the note.

**Comment by robin-a-meade** [comment]
**Tip: Create alert boxes**

Paste the following into a new VNote note to see a demo of a way to create alert boxes:
```
# Demo of Alert Boxes

This demonstrates a way to produce alert boxes.

<div class=NOTE></div>

> **NOTE**: This blockquote will be styled using the CSS NOTE class defined below.
> I found the technique [here](https://developer.run/17).

<div class=DANGER></div>

> Slippery when wet

<style>
.NOTE+blockquote {
padding: 8px 35px 8px 14px;
margin-bottom: 20px;
border-radius: 4px;
background-color: #d9edf7;
border: 1px solid #bce8f1;
}
.NOTE+blockquote p {
color: #31708f;
}
.DANGER+blockquote {
padding: 8px 35px 8px 14px;
margin-bottom: 20px;
text-shadow: 0 1px 0 rgba(255,255,255,0.5);
background-color: rgba(232,76,61,0.1);
border-radius: 4px;
border: 1px solid rgba(232,76,61,0.1);
}
.DANGER+blockquote p {
color: #d82a1a;
}
.DANGER+blockquote p:before {
content: "⚠ ";
font-weight: bold;
font-size: larger;
display: inline;
}
</style>
```
**Screenshot showing result in VNote preview mode:**

![image](https://user-images.githubusercontent.com/1343270/38773557-44afe840-3feb-11e8-811e-1e7c2a924525.png)

This is a good approach because markdown compatibility is maintained; these alerts will degrade gracefully into plain blockquotes when processed by classic markdown.

With use of jQuery (or similar), the marker div wouldn't even be needed. A jQuery script could search for all blockquotes beginning with `<b>NOTE</b>` and add the INFO class. This would simplify the CSS selectors too.

A next step would be to try to incorporate the CSS (and maybe jQuery script) into a theme. I haven't looked into that yet.

**Comment by LeoSantos7** [comment]
I've downloaded VNote especially to check its vim mode. How do I enter the VIM mode?

**Comment by tamlok** [comment]
@LeoSantos7 Hi, please enable Vim mode in the settings dialog and restart VNote.

**Comment by light42** [comment]
Hi @tamlok, I love your application. But I hesitated to fully commited using vnote, because I'm not sure if it could handle searching large volume of notes. I need to know, how many notes vnote could save until it stopped to behave properly, ten thousand, a hundred thousand? Because I'm really a messy person, and need a very long time before I could process my raw notes into pretty essay and throw it off.

**Comment by tamlok** [comment]
@light42 VNote stores your notes in native file system, so VNote could handle as many notes as your OS could handle. Load of notebooks/folders/notes is on demand. When doing full-text search, you could restrict the search within opened notes, current folder, current notebook, or all notebooks. The search is implemented via C++. I think it is fast enough even for thousands of files (or at least faster than those Electrons). 

BTW, when you have a hundred thousand files, I think the way to organize them is more important.

Thanks!

**Comment by lisdude** [comment]
I definitely appreciate all of your work! VNote has finally ended my ever growing Markdown document comparing note taking software. 

I don't really have anything constructive to say other than thank you and, because it's 2019, an emoji: 👍

**Comment by watson387** [comment]
Great editor/viewer/notebook you got here.  I've tried a lot of them and I've finally settled on Vnote.  It has an intuitive interface and the fact that it saves everything in plain text/MD and helps organize folders is a major bonus.  I use it for everything from a work diary, notes on my new house, Postgresql schemas and scripts, to-dos, etc. Thanks for this tamlok!

Edit: I also want to note that I use it on two Windows computers and two computers running Ubuntu Disco.  I haven't come across any of the strange behaviors in Windows so far.

**Comment by watson387** [comment]
[Markvis](https://github.com/geekplux/markvis) is a markdown-it plugin for creating graphs.  Is this something that would be difficult to add to Vnote?  mermaid is great for flowcharts.  Having something to create graphs would be excellent.

**Comment by henices** [comment]
I hope we can have a upgrade notifiation.

Vnote 2.7.2 is avaliable.   :)

**Comment by lisdude** [comment]
If you're looking for suggestions for new features, could we have a folder monitor that would automatically add new markdown files as notes? I'm trying out a workflow where I use 1Writer for iOS to manage my notes when I'm away from my laptop, but when I add new files with 1Writer I have to manually import them into VNote. It would be handy if VNote would just see that the file was added to a monitored directory and auto-import it as a note.

**Comment by tamlok** [comment]
@lisdude Try the Explorer tab at the top left
corner? In this way, you will use vnote as an editor with full text search.

**Comment by versipellis** [comment]
What's the difference between quick and flash notes?

**Comment by tamlok** [comment]
@versipellis Hi, flash page is a temp page for quick notes, draft, or flash ideas. Quick access page is a frequently accessed note you specify, with which you could quickly open it (for example, a note with frequently used commands or scripts).

**Comment by versipellis** [comment]
@tamlok What is the actual functional difference between the two, though? Both seem to behave the same way.

**Comment by tamlok** [comment]
@versipellis Just two different ways to call out given notes handily.

**Comment by tamlok** [comment]
Hi,

To rename a notebook, just pop up the combo box of notebook list, and then
right click a notebook button.

BTW, please post a new issue for usage issues about VNote.

Regards,
Le

在 2019年9月5日星期四，Ben Tang <notifications@github.com> 写道：

**Comment by tamlok** [comment]
@neowood Hi, you could search notes via `Universal Entry` (`Ctrl+G`) and close unnecessary panels from the `View` menu.

**Comment by tamlok** [comment]
@neowood `Ctrl+E E` to expand the edit area. There is no need to pin the `Universal Entry`, just call it out via `Ctrl+G` each time you need it.

**Comment by mika-cn** [comment]
My fault. My Vnote was too old.

**Comment by ibnishak** [comment]
Love your work man. Absolutely great! I think I finally found the perfect way to manage my personal knowledge base.

**Comment by skyblue002** [comment]
I really love it so far! Thank you for taking the time to create it. I'm especially glad that you included support for footnotes. I really need them for my writing and the other markdown editors I've tried don't support them. I also really love the sidebar (with the Notebooks list and the Notes list). There are so many markdown editors that don't have that feature. I really love having it around. It allows me to see where I am in a large project at a glance and easily refer to other Notes if I need to. I also really like that the write/preview modes are in the same window and you just press a button to toggle them. I prefer that over having two separate windows. It's really a great app. I'm so glad I found it! :)

**Comment by Agrotsa** [comment]
Can we contribute with translations? Even if my system language is other, I have Vnote in english and the settings panel has only chinese and japanese to choose.

**Comment by louwers** [comment]
I'd love to chat about vnote but the Slack link in the README and on the homepage don't work anymore!

**Comment by tamlok** [comment]
@Agrotsa Sorry for the late reply. Just saw your comments. By default, VNote should appears as English. I think I need to add an explicit option for English. Thanks!

**Comment by tamlok** [comment]
@louwers It works fine on my side. Could you join the workspace directly via search?

![image](https://user-images.githubusercontent.com/5763157/71858551-a1b91100-3126-11ea-8cbf-dd3a8bd6b1ef.png)


I am not familiar with Slack.

Just updated the share link: https://join.slack.com/t/vnote/shared_invite/enQtNDg2MzY0NDg3NzI4LTVhMzBlOTY0YzVhMmQyMTFmZDdhY2M3MDQxYTBjOTA2Y2IxOGRiZjg2NzdhMjkzYmUyY2VkMWJlZTNhMTQyODU

**Comment by louwers** [comment]
Weird, I get this with the link in the README.

![image](https://user-images.githubusercontent.com/649392/71877093-a5e62e00-3128-11ea-9c01-41e8e1fe6571.png)

But the link in your comment works.
Thank you. :+1:

**Comment by Agrotsa** [comment]
> @Agrotsa VNote should appears as English. I think I need to add an explicit option for English. Thanks!

I think that I was a bit confusing in mi message, sorry. What I meant is that would be nice if the users could contribute translating Vnote to our respective languages, because so far Vnote it's only available in english, chinese and japanese.
Many other FLOSS projects have an account in Weblate, Crowdin, or similar platforms so their users can collaborate translating the text chains. For example, if I dont recall wrong, I think that Qownnotes, a similar project to Vnote, uses Crowdin. But the concrete platform is the least as long as there's any "system" to permit the users collaborate in translating, ergo spreading, Vnote. I just put those platforms as examples.
Have you thought about that?

**Comment by tamlok** [comment]
@Agrotsa OK. I will give it a try.

**Comment by f0nzie** [comment]
This is an A.W.E.S.O.M.E. application. Its workflow and search capabilities is far superior than Evernote. I already recommending it in Twitter to the #rstats community (data science), always looking for ways of keeping reproducibility high.

Up to this moment I had my markdown notes in folders without any "note management". My editor is Typora. I moved all my notes to vnote and they are far better organized making sense of all that data.

There are small rough edges here and there, which I am finding by using it in production. For instance I found that "open in external editor" only works when you move the cursor in the note and then press F12; but F12 doesn’t work in the editor; images links are pasted correctly but when you size them the code at the end (suffix), example "=500”, is not compatible with Typora markdown or Github markdown. There are couple more annoyances that I would like to fix with pull requests.

Do you have instructions to develop and build in Linux Debian? That would be great for submitting collaboration.

Great work!

**Comment by tamlok** [comment]
@f0nzie

Hi,

Thank you for your attention!

1. The image size way `=500` is not a standard Markdown syntax. It is just an extension of `markdown-it` parser. So it is not surprised that it does not compatible with Typora and GitHub markdown. (BTW, Typora uses a more "unique" markdown syntax as far as I know.)
2. You could find help of development in the [documentation of VNote](https://tamlok.gitee.io/vnote/en_us/#!docs/Developers/Build%20VNote.md).

**Comment by f0nzie** [comment]
@tamlok 
Thanks.

I just built VNote from source under Ubuntu 18.04. All went well. I had only to install one dependency with `sudo apt-get install libgl1-mesa-dev`. Qt 5.9.7 is installed by default by Ubuntu 18.04. I did a quick `qmake -v` to find out.

Typora uses as a base GitHub markdown. Typora adds few more additions such as the ability of doing highlighting text with `==`, and footnotes. It is a good thing to have the best of both worlds at hand: VNote for note taking and organization, and Typora (or markdown editor of preference), for a quick WYSIWYG.

Would it help if I build VNote binary for Debian 64-bit?

By the way, what is the C++ editor you would recommend to develop VNote?

**Comment by tamlok** [comment]
@f0nzie Now there is no package for Ubuntu (as well as Snap) or Debian, so it would be appreciated if you could help packaging VNote on different Linux distros.

I use Vim as well as QtCreator (for the documentation) when developing VNote.

**Comment by f0nzie** [comment]
Ok. I will give it a try first with Ubuntu. And then progress from there.

This afternoon I was able to build VNote with **QtCreator**. No hicups. Perfect.
I tried also with `vscode` but it is complicated to set up `cmake`, etc.
QtCreator works well.

Now, I am getting familiar with the code. It will take some time to understand before I dare to make any changes or additions. For instance, I would like to start by adding the button **F12** (open with system's default program) to the **Editor**. Which code file would you suggest to start taking a look to implement that change?

**Comment by tamlok** [comment]
@f0nzie You could check out `VEditArea::registerCaptainTargets()` or `VEditArea::initShortcuts()`. If you have other questions, feel free to open a new issue. :)

**Comment by RyanGreenup** [comment]
> Try the Explorer tab at the top left corner? In this way, you will use vnote as an editor with full text search.

What is the practical distinction between using a notebook and just using the editor view? If I had a few MD files in folders I want to import is it worth going through the effort of recreating the folders inside Vnote manually?

**Comment by tamlok** [comment]
@RyanGreenup If you won't change the structure of your notes outside from VNote frequently (for example you are using other MD editors at the same time), I suggest you could use notebook of VNote (let VNote manage all the stuffs). Otherwise, using VNote as an editor is more convenient.

**Comment by versipellis** [comment]
Is there a way to bullet a bunch of lines at the same time? I often copy+paste in notes from lecture powerpoint slides that are bulleted, and they come in as individual lines. Having to individually add `* ` to every line copied in is very tedious. A lot of times they come in with trailing white spaces too.

**Comment by tamlok** [comment]
@versipellis Hi, there is no such function currently and will add it as a plugin in the future. BTW, please post a new Issue for such issues. :)

**Comment by sapphire008** [comment]
Thank you for creating this wonderful cross platform markdown note taking app!

**Comment by Bennett-Yang** [comment]
I really appreciate you for creating this awesome app. I'm wondering why the 'cart' gets cleared as long as I close VNote. What's the point of this design? I think keeping it persistent may be better.

**Comment by tamlok** [comment]
@Bennett-Yang The `Cart` is a straightforward way to process multiple notes together at once. I think `Pin to History` or tags will be more useful in your case? Anyway, will write it into a config file in the future. Thanks!

**Comment by Bennett-Yang** [comment]
@tamlok Thank you! I think I'm already addict to write with VNote. Hope it become better and better!

**Comment by jbcaramar** [comment]
I just spent the last week looking for Linux tools that would allow me to easily organize and maintain a collection of notes and documentation using markdown. Although there's some nice apps for markdown out there, none of them seem to be designed primarily for use as a note manager per se, The ones that try often hide your markdown in a sqlite db, which is not terribly convenient.

I didn't find mention of VNote anywhere until two days ago (why is VNote not more widely known?) . It's exactly what I've been looking for! Notes open as rendered markdown, their appearance is easily customizable, there's sidebar navigation for organizing your notes, you can use tags... it's like everything I wanted in a single app.

Thank you, thank you, thank you for making such an excellent application! I switched from Mac OS to Linux a couple of years ago and this is one of the most brilliant apps I've run into on Linux so far. Using markdown for notes had been problematic, so I decided that I needed to get serious about finding a working solution. I think finding VNote just resolved all my issues at once!

Please keep up the great work on this app!

**Comment by RyanGreenup** [comment]
Yeah VNote is arguably the best app for note taking (other than`org-mode ` if you have the patience to configure emacs) but it’s so unheard of compared to things like Notable and boostnote. 

@jbcaramar You should look at installing `recoll ` , it will let you build a search engine for the markdown notes that you can search instantly, in conjunction with interlinking notes you’ll feel so organised.

**Comment by irosyadi** [comment]
Thank's for your great software @tamlok . It is unfortunate I just found VNote lately. I've compared it to Zettlr or Obsidian, and I think VNote is far better.

**Comment by ribezzz** [comment]
hi, 
thank you for your program!

There is a way to include [font awesome](https://fontawesome.com/) icons?

**Comment by tamlok** [comment]
@ribezzz thank you! please post a new issue for the font awesome.

**Comment by proxseas** [comment]
@tamlok I'm very impressed with this application but there is one glaring downside - table management. As far as I can tell, tables are treated as plaintext. This is a very common shortcoming for many markdown editors - but not all of them. Typora, MarkText, and Joplin (in WYSIWYG mode) all support basic table management operations like inserting/deleting rows/columns.
Furthermore, viewing tables in VNote's particular manner is not a pleasant experience. I can always toggle out of editing into the rendering mode, but then I can't edit the table anymore. Is it possible, by chance, to view the output/rendered version of the document, side by side, as many markdown editors do? I tried splitting the content area and opening another instance of the same file, but VNote didn't open it (deliberately, I'm sure).
Everything is top notch in VNote. This is my only gripe!

**Comment by tamlok** [comment]
@proxseas VNote now will automatically align columns of table but does lack shortcut to add line or column. Please check out this video for the smart table feature. Will enhance it someday.

https://www.youtube.com/watch?v=EFCmCFZKxJ4

**Comment by JasonAn** [comment]
in the preview mode. how can i soft wrap the code block? 
![image](https://user-images.githubusercontent.com/4537312/88231875-c6d82180-cc29-11ea-98ae-a2b06a2f8a83.png)

**Comment by tamlok** [comment]
> in the preview mode. how can i soft wrap the code block?
> ![image](https://user-images.githubusercontent.com/4537312/88231875-c6d82180-cc29-11ea-98ae-a2b06a2f8a83.png)

You need to customize the CSS. Refer to the documentation for details.

P.S. For such issues, please post a new issue instead of here. :)

**Comment by RyanGreenup** [comment]
I use Vnote with a collection of other tools like vim/emacs/vscode/marktext and so on. When I create new files outside Vnote they aren't tracked and I find myself having to delete all the json files and then make a new notebook which will rescan all the files and directories anyway:

```bash
find ./ -name '_vnote.json' -exec rm {} \; 
```

I can't just use the explorer mode either because it doesn't follow links inside VNote.

I think it would be better if Vnote had:

- some capacity to watch files, or
- An option to recursively scan all files in a directory for new files, or
- An option to delete all the `_vnote.json` files and choose a new directory (even if it's the same directory)
- follow links to MD files in Explorer mode?

what do you think?

**Comment by sakkamade** [comment]
@tamlok, Hello. Thank you, for great application.

Is there's a way to make the `<script>` tag to work? 

Like to insert `JS` between `<script>` and `</script>` tags.

**Comment by feloxx** [comment]
Qun Zhu Niu Bility

**Comment by tamlok** [comment]
@sakkamade Hi, the `<script>` tag should work but you need to hook it to some event to trigger. Feel free to open a new issue for this.

**Comment by Yu-Shengi** [comment]
windows 10 安装 VNote 3.4.0 为什么很多时候刚刚打开就会直接卡死

**Reply by tamlok** [reply]
> windows 10 安装 VNote 3.4.0 为什么很多时候刚刚打开就会直接卡死

https://github.com/vnotex/vnote/issues/853

**Reply by tamlok** [reply]
如果是跨版本更新，删除用户文件夹下面的vnotex.json

**Comment by pengchenggang** [comment]
希望能有一个删除当前行的快捷键 我用vscode，或者是idea，就是用ctrl+d 删除当前行，非常好用，或者咱们有这个功能也行哟

**Reply by tamlok** [reply]
VNote 目前只有 Vi 的按键模式，后面会添加 vscode 的模式。

**Reply by tamlok** [reply]
最新版本已经添加vscode的按键模式了

**Comment by GeShiDeTianXin** [comment]
大佬，能不能加个双链的功能

**Reply by tamlok** [reply]
目前正在重构笔记管理模块，会有双链的

**Comment by JoeJoeTV** [comment]
Not sure if this is something exclusive to me, but VNote 4 seems a bit sluggish in comparison to VNote 3. What I mean is when I'm typing it feels like there is a greater delay between me typing and the input actually appearing in the editor.
Is not really that big of an issue and except for that I've been enjoying the new version, just a small nitpick.

Also for some reason when I just type `- ` (including the space) and press Enter, the text disappears like I didn't type it. IIRC in VNote 3 this would just go to the next line.
Is this configurable? I usually use this when I lay out the sections of a document and start a list in each one.

**Reply by tamlok** [reply]
> Not sure if this is something exclusive to me, but VNote 4 seems a bit sluggish in comparison to VNote 3. What I mean is when I'm typing it feels like there is a greater delay between me typing and the input actually appearing in the editor. Is not really that big of an issue and except for that I've been enjoying the new version, just a small nitpick.
> 
> Also for some reason when I just type `- ` (including the space) and press Enter, the text disappears like I didn't type it. IIRC in VNote 3 this would just go to the next line. Is this configurable? I usually use this when I lay out the sections of a document and start a list in each one.

Thanks @JoeJoeTV ! Will have a check and fix. For the typing delay, could you check editing a txt file to see if there is still the same delay.

**Reply by JoeJoeTV** [reply]
Ok, so I did try this with a text file and also again with a Markdown file and I'm not sure if it was something else, but at the moment I can't reproduce the delay I was experiencing (this is still not with the newest version). Also with the new version, I couldn't reproduce it.

But the behaviour of deleting the line when hitting enter after ` -` is still there.

**Reply by tamlok** [reply]
> Ok, so I did try this with a text file and also again with a Markdown file and I'm not sure if it was something else, but at the moment I can't reproduce the delay I was experiencing (this is still not with the newest version). Also with the new version, I couldn't reproduce it.
> 
> But the behaviour of deleting the line when hitting enter after ` -` is still there.

`- ` will be deleted when hitting Enter is expected, which means cancelling a list.

You could hit `Shift-Enter` to avoid this behavior.

**Reply by JoeJoeTV** [reply]
IIRC, this was not that way in VNote 3, but would have to check.

Is there a way to reconfigure this, so it just keeps an empty list entry?

if not, that's fine, just wanted to confirm.

**Reply by tamlok** [reply]
> IIRC, this was not that way in VNote 3, but would have to check.
> 
> Is there a way to reconfigure this, so it just keeps an empty list entry?
> 
> if not, that's fine, just wanted to confirm.

No config. Just Shift+Enter
